sbi/inference/posteriors/base_posterior.py

```diff
@@ -31,13 +36,29 @@ def __init__(
     ):
         """
         Args:
-            potential_fn: The potential function from which to draw samples.
+            potential_fn: The potential function from which to draw samples. Must be a
```

Comment:
We might want to adapt the type set above to `Union[BasePotential, Callable]`? 
Not sure which downstream effects this will have with `pyright` issues..

Reply:
Agreed! Change the type hint, no pyright issues :)